### PR TITLE
feat(v2): Integrate v2 CRDA API

### PR DESCRIPTION
**Pull request by mathur07**
Changes related to v2 stack analysis

### CODE REVIEW

File: src/multimanifestmodule.ts [hunk 1]
```diff
@@ -8,21 +8,23 @@ import { stackanalysismodule } from './stackanalysismodule';
...
         .then(item => {
           let form_data = {
-            'manifest[]': [],
-            'filePath[]': [],
-            'license[]': [],
-            origin: 'lsp'
+            'manifest': '',
+            'file_path': '',
+            'license[]': '',
+            origin: 'lsp',
```
Comment (arajkumar) [inline src/multimanifestmodule.ts:22]: There is no origin field in the form data. It should be removed.
```suggestion
```

File: src/stackanalysismodule.ts [hunk 2]
```diff
@@ -46,12 +46,11 @@ export module stackanalysismodule {
...
               options['uri'] = `${
                 Apiendpoint.STACK_API_URL
-              }stack-analyses/?user_key=${Apiendpoint.STACK_API_USER_KEY}`;
+                }stack-analyses/?user_key=${Apiendpoint.STACK_API_USER_KEY}`;
               options['formData'] = payloadData;
               options['headers'] = {
                 showTransitiveReport: 'true',
-                origin: 'vscode',
-                ecosystem: Apiendpoint.API_ECOSYSTEM
+                'x-3scale-account-secret': 'not-set'
```
Comment (arajkumar) [inline src/stackanalysismodule.ts:53]: `x-3scale-account-secret` is no longer needed as we are dealing with staging data
```suggestion
```

File: src/stackanalysismodule.ts [hunk 3]
```diff
@@ -68,9 +67,12 @@ export module stackanalysismodule {
...
               options['uri'] = `${
                 Apiendpoint.STACK_API_URL
-              }stack-analyses/${respId}?user_key=${
+                }stack-analyses/${respId}?user_key=${
                 Apiendpoint.STACK_API_USER_KEY
-              }`;
+                }`;
+              options['headers'] = {
+                'x-3scale-account-secret': 'not-set'
+              };
```
Comment (arajkumar) [inline src/stackanalysismodule.ts:75]: `x-3scale-account-secret` is no longer needed as we are dealing with staging data
```suggestion
```

File: src/stackanalysismodule.ts (hunk 2 above)
Comment (mathur07) [reply src/stackanalysismodule.ts:53]: Are we not making these changes for production data?
For the release of free-tier?

File: src/stackanalysismodule.ts (hunk 3 above)
Comment (mathur07) [reply src/stackanalysismodule.ts:75]: Are we not making these changes for production data?
For the release of free-tier?

File: src/multimanifestmodule.ts (hunk 1 above)
Comment (mathur07) [reply src/multimanifestmodule.ts]: fixed

File: src/authextension.ts [hunk 4]
```diff
@@ -7,11 +7,11 @@ export module authextension {
...
 
   setContextData = (context_f8_access_routes, context_f8_3scale_user_key) => {
-    Apiendpoint.STACK_API_URL = context_f8_access_routes.prod + '/api/v1/';
+    Apiendpoint.STACK_API_URL = context_f8_access_routes.prod + '/api/v2/';
     Apiendpoint.STACK_API_USER_KEY = context_f8_3scale_user_key;
     Apiendpoint.OSIO_ROUTE_URL = context_f8_access_routes.prod;
     process.env['RECOMMENDER_API_URL'] =
-      context_f8_access_routes.prod + '/api/v1';
+      context_f8_access_routes.prod + '/api/v2';
     process.env['THREE_SCALE_USER_TOKEN'] = context_f8_3scale_user_key;
```
Comment (dvandra) [inline src/authextension.ts:15]: The User token should be updated with respect to stage for v2 api.

File: src/extension.ts [hunk 5]
```diff
@@ -152,7 +152,7 @@ export function activate(context: vscode.ExtensionContext) {
 
   let showInfoOnfileOpen = (msg: string) => {
     vscode.window
-      .showInformationMessage(`${msg}.`, 'Dependency Analytics Report ...')
+      .showInformationMessage(`${msg}. Powered by [Snyk.io](https://snyk.io)`, 'Dependency Analytics Report ...')
```
[the remark this answers is not in the document]
Comment (mathur07) [reply src/extension.ts:155]: Okay, I will fix this.

File: src/stackanalysismodule.ts (hunk 2 above)
Comment (arajkumar) [reply src/stackanalysismodule.ts:53]: It was meant for devcluster, not required for staging or prod.

File: src/stackanalysismodule.ts (hunk 3 above)
Comment (arajkumar) [reply src/stackanalysismodule.ts:75]: It was meant for devcluster, not required for staging or prod.

File: src/authextension.ts (hunk 4 above)
Comment (arajkumar) [reply src/authextension.ts:15]: What is that user token @dvandra ?

File: src/authextension.ts [hunk 6]
```diff
@@ -7,11 +7,11 @@ export module authextension {
   export let setContextData: any;
 
   setContextData = (context_f8_access_routes, context_f8_3scale_user_key) => {
-    Apiendpoint.STACK_API_URL = context_f8_access_routes.prod + '/api/v1/';
+    Apiendpoint.STACK_API_URL = context_f8_access_routes.prod + '/api/v2/';
     Apiendpoint.STACK_API_USER_KEY = context_f8_3scale_user_key;
     Apiendpoint.OSIO_ROUTE_URL = context_f8_access_routes.prod;
     process.env['RECOMMENDER_API_URL'] =
-      context_f8_access_routes.prod + '/api/v1';
+      context_f8_access_routes.prod + '/api/v2';
```
[the remark this answers is not in the document]
Comment (arajkumar) [reply src/authextension.ts:14]: That change should be made in a separate PR so that we can revert it before promoting to production.

File: src/authextension.ts (hunk 4 above)
Comment (dvandra) [reply src/authextension.ts:15]: I mean to say the `f8_3scale_user_key` for stage.

Comment (dvandra) [reply src/authextension.ts:15]: According to this : https://github.com/fabric8-analytics/fabric8-analytics-vscode-extension/pull/375#discussion_r443338028

It looks good to me.

**Comment by mathur07** [comment]
[test]

File: src/extension.ts [hunk 7]
```diff
@@ -152,7 +152,7 @@ export function activate(context: vscode.ExtensionContext) {
 
   let showInfoOnfileOpen = (msg: string) => {
     vscode.window
-      .showInformationMessage(`${msg}.`, 'Dependency Analytics Report ...')
+      .showInformationMessage(`${msg}. Powered by [Snyk](https://app.snyk.io/signup/?utm_medium=Partner&utm_source=RedHat&utm_campaign=Code-Ready-Analytics-2020&utm_content=Register)`, 'Dependency Analytics Report ...')
```
Comment (arajkumar) [inline src/extension.ts:155]: Again, the utm has to be kept in a common file, probably constants.ts might be the best file.

**Comment by arajkumar** [comment]
[test]

**Comment by mathur07** [comment]
[test]

**Comment by arajkumar** [comment]
[test]

**Comment by arajkumar** [comment]
[test]

**Comment by arajkumar** [comment]
[test]

**Comment by arajkumar** [comment]
@mathur07 , few tests are failing. Please fix it.

https://ci.centos.org/job/devtools-fabric8-analytics-vscode-extension/500/console

**Comment by mathur07** [comment]
[test]

**Comment by mathur07** [comment]
[test]

**Comment by dvandra** [comment]
[test]

File: package.json [hunk 8]
```diff
@@ -2,7 +2,7 @@
   "name": "fabric8-analytics",
   "displayName": "Dependency Analytics",
   "description": "Insights about your application dependencies: Security, License compatibility and AI based guidance to choose appropriate dependencies for your application.",
-  "version": "0.0.13",
+  "version": "0.0.14",
```
Comment (arajkumar) [inline package.json:5]: Do it in a dedicated PR., not as part of this.

**Comment by arajkumar** [comment]
[test]

**Comment by mathur07** [comment]
@arajkumar No, tests are not failing in my local now..

File: package.json (hunk 8 above)
Comment (mathur07) [reply package.json]: fixed it

**Comment by mathur07** [comment]
[test]

**Comment by mathur07** [comment]
[test]

**Comment by arajkumar** [comment]
[test]

**Comment by arajkumar** [comment]
@mathur07 , you must add yourself into `admin-list` from `app-interface/resources/jenkins/ci-centos/defaults.yaml` to trigger test jobs.
